### ISSUE TITLE
MSVCR120.DLL version

### ISSUE THREAD

**Issue by Amanoo** [Hollowscope4022/Playstation-Plus-linux#1]
Your instructions don't say which version of MSVCR120.DLL to download. The download link just points to the main page which lists different versions in both x64 and x86. I'm currently going by the assumption that I need the latest 32 bit version but I have yet to go through all the instructions so I don't know if that's the correct version yet.

**Comment by Hollowscope4022** [comment]
You're gonna want to download the 32 bit version, it's the first one at the top of the list, sorry I forgot to include that in the instructions. They'll be updated soon.

**Comment by Amanoo** [comment]
Great! 

I got it working via your method now. Somehow it wouldn't using other methods. I did notice the white square in the bottom left corner is still there. My current workaround is to switch to my second workspace (on my system only the main monitor has multiple workspaces, the side monitors only have 1), open PlayStation Plus, then move it to a side screen, go back to workspace 1, and move it back to my main screen. The white square is now on workspace 2 and my PS+ and any games I may want to play on workspace 1. Just move PS+ to another workspace, in short.

On another note, have you gotten the DualShock 4 controller working? I tried playing Spider-Man Miles Morales, but I'm now stuck because the game insists I should use the controller's touchpad. I have a semi-official controller from Nacon with a touchpad but it doesn't register the touchpad. Steam also thinks it's a DS3 controller.

**Comment by Hollowscope4022** [comment]
I got everything on my ps5 controller to work except the TouchPad and adaptive triggers , there are some games that insist the use of the touchpad like miles morales, I've played that too and can't switch suits because of it. I've looked for a way to get it mapped or working but all I can get it to do is work like a laptop TouchPad, there's not that great of documentation on the ps5 controller for Linux imo, I'll keep looking for a fix and show it here if I find it because this is affecting me as well.

**Comment by Amanoo** [comment]
You could try to see if it will work if you launch it through Steam or something. You may have to enable PlayStation controller support first. It should be able to understand DualShock 4 and maybe also 5? Definitely should understand 4, I've seen screenshots of Steam recognising it as such. Although my DS4 is only detected as a DS3, without the touchpad. My working theory right now is that it's not an actual Sony, just licenced by Sony, and that an actual Sony DS4 may give me better luck. Maybe the device registers slightly different because if that, it probably at least has a different vendor ID if nothing else. But even if I can get Steam to understand my controller properly, I still don't know if I can get PlayStation Plus to understand it.

Definitely do keep me posted if you find a workaround. Even if it's something silly like using a mouse or a trackpad, or a second controller, hell, even a Wacom tablet.

**Comment by Hollowscope4022** [comment]
Launching it through steam makes it crash a lot, and for some reason try to install a ps plus update which I've never seen it do before even on windows. Steam might be able to recognize the TouchPad but idk if ps plus will, also I might look into seeing what spiderman remastered uses to work with the TouchPad, steam prompts you to get some sorta firmware update for the ps5 controllers but nothing else does, I might see if me updating my controller firmware resolves this or if I can snag a direct input dll from spiderman remastered

**Comment by Amanoo** [comment]
Snatching a dll from Spider-Man Remastered is an interesting idea. I'm not sure how it could then be used by WINE, but then again I only have a vague understanding of what dll files even are. They don't really exist in my branch of IT.

**Comment by Hollowscope4022** [comment]
I got the controller firmware updater running, I just need to charge my controller for a bit before it will alow me to update it

**Comment by Hollowscope4022** [comment]
The controller firmware updater won't update my controller because it constantly thinks it's not charged enough, itv'e been charging it nonstop wince my last comment. Any luck with your steam remapping?

**Comment by Amanoo** [comment]
No luck with Steam. It thinks it's a PS3 controller. My working hypothesis is because it's a different brand than Sony.

**Comment by Amanoo** [comment]
I just checked, I have one of very few PS4 controllers that have some Xinput shenanigans going on that mean the PC can't fully detect the controller's feature set. Apparently it only supports XInput when connected to a PC or something.

~~I have found this project: https://github.com/r57zone/DualShock4-emulator It's not the greatest solution if you do actually have a touchpad and it's just refusing to work, but I'm more used to playing Spider-Man on my GameCube style controller anyway. It's meant for Windows though, so I'm not sure if it will work in WINE.~~ I was a bit hasty in writing this. Doesn't work in WINE.

**Comment by Hollowscope4022** [comment]
I'm picking up an 8bitdo controller I ordered in the mail a while back, and I got the firmware for the ps5 controller to install, so I'll let you know if any of those things are a solution to this.

**Comment by Hollowscope4022** [comment]
There's also ds4drv and with that you can type  evtest into the terminal and I guess manually map your controls, it looks rather complicated but I'm also gonna take a bigger look into it. Also I checked spiderman remastered for any dll files that might work for the controller and I didn't find any that looked substantial.

**Comment by Amanoo** [comment]
I got an official DualShock 4 v2. Linux immediately knew what to do with it. Touchpad worked as a mouse, Steam detected it as a DualShock 4. I tested three other controllers and all invariably were detected as something else. The Hori Onyx Plus has a PS4 mode but that wouldn't work at all.

I haven't been able to figure out how to use the touchpad in PlayStation Plus yet. My computer at least knows it's there, but PS+ doesn't. I tried running PS+ through Steam but that didn't help stability. I know wineusb exists but I haven't found how to use it that and it might be too early in its development. I might still have to use a VM for Spider-Man Miles Morales.

**Comment by Etaash-mathamsetty** [comment]
don't download msvcr120 from a sketchy website, try using `winetricks -q vcrun2013` to install it

**Comment by Whayme** [comment]
> don't download msvcr120 from a sketchy website, try using `winetricks -q vcrun2013` to install it

Unfortunately, the one that grabs is much smaller and doesn't work.  Agreed on not using dll sites though.  It's one thing to take the risk yourself, but it's kinda irresponsible to suggest it to others.

**Comment by Etaash-mathamsetty** [comment]
> > don't download msvcr120 from a sketchy website, try using `winetricks -q vcrun2013` to install it
> 
> Unfortunately, the one that grabs is much smaller and doesn't work. Agreed on not using dll sites though. It's one thing to take the risk yourself, but it's kinda irresponsible to suggest it to others.

true, but could you attach a log with WINEDEBUG=+msvcrt, or submit a wine bug, so that we don't have to download a dll from a sketchy website. We should try getting this bug fixed before wine 8.0
